### DISCUSSION TITLE
State Changes broken on iOS17 ?

### DISCUSSION THREAD

**Discussion by Aytharn** [fsprojects/Avalonia.FuncUI#378] [General]
in iOS17 you can load UIs but never change their states, while in the rest of the xplat project platforms everything runs smooth

**Comment by JaggerJo** [comment]
Please provide more details. For bugs I always expect:

- a code snippet / optimally a project I can use to reproduce the issue
- all your versions (Avalonia Version, FuncUI Version, .NET Version, ...) 

Otherwise it's just pointless for anyone to even look at this.

**Comment by Aytharn** [comment]
https://www.swisstransfer.co*5-47e8-b4b4-c3ec17913146
there you go

**Reply by JaggerJo** [reply]
from looking at the sample you are using Avalonia version: `11.1.999-cibuild0042491-beta` Have you tried it with `11.0.5`/`11.0.6` yet?

**Reply by JaggerJo** [reply]
@Aytharn nice looking app btw.

**Reply by Aytharn** [reply]
Yes of course, I tried them
Both, but I need nightly now. Since to convert my software I need EF ComplexTypes, hence dotnet/efcore 8 as a minimal prerequisites…

in this sample I stripped everything to highlight my 3 problems.

1 - state doesn’t change on iOS17 simulator, on device I didn’t try 
2 - Lottie isn’t loading AvaloniaResources, it’s properly included, but nothing is displayed, you can see on login.fs
3 - in doctors.fs you can clearly see how the focus doesn’t work and if it would, it does crash on dispose 

PS: thanks, I try to make it look modern yet familiar with the previous software.

**Reply by JaggerJo** [reply]
If you want me to take a look at a specific issue you need to provide a **MINIMAL** demo project I can run. 

If your company needs 1:1 support reach out to me via mail and we'll figure out a support contract.

**Comment by JordanMarr** [comment]
Aytharn, 

Do you have the same issue with the provided [iOS example](https://github.com/fsprojects/Avalonia.FuncUI/blob/master/src/Examples/Examples.Mobile.iOS/Program.fs)?

**Reply by Aytharn** [reply]
I was working ok, but I was trying with iOS16

**Reply by JaggerJo** [reply]
It would make sense to try the [iOS example](https://github.com/fsprojects/Avalonia.FuncUI/blob/master/src/Examples/Examples.Mobile.iOS/Program.fs) on iOS 17 and see if the issue is caused by your code.

**Reply by Aytharn** [reply]
https://github.com/fsprojects/Avalonia.FuncUI/assets/8307825/fa228fed-a53c-474d-af25-118e82372631

This way? It doesn't change state!
My code was working beautifully under iOS16 as it is under android or macOS

**Reply by JaggerJo** [reply]
Ok 👍 So now we know the issue is not caused by your code.

I'm taking a look.

**Reply by JaggerJo** [reply]
<img width="1315" alt="image" src="https://github.com/fsprojects/Avalonia.FuncUI/assets/13090415/443eee88-36a5-4029-8118-095c184d8679">

Which versions do you use?

The following work for me:
dotnet: `net8.0-ios`
FuncUI: `master` / `1.1.0`
Avalonia: `11.0.0`

I can reproduce your bug on Avalonia 11.0.5, 11.0.4, 11.0.3, 11.0.2, 11.0.1.

**Reply by JaggerJo** [reply]
Just tried 11.0.6 and it's also broken.

**Reply by Aytharn** [reply]
And the Lottie animation? does it show anything on 11.0.0 ?
My guess is that is has nothing to do with it but with the fact that the animation is embedded as an AvaloniaResource, which was untested

**Comment by Aytharn** [comment]
just in your example you mention opening Avalonia.iOS library which I don't have... is it deprecated or in another nuget?

**Reply by JaggerJo** [reply]
it's another nuget from avalonia.

**Reply by Aytharn** [reply]
I see it's already in its separate .iOS project

**Comment by Aytharn** [comment]
11.1.999 nightly

**Reply by maxkatz6** [reply]
Please try with nightly builds from this PR 
https://github.com/AvaloniaUI/Avalonia/pull/13942

Should be fixed

**Reply by Aytharn** [reply]
Almost, the state change bug is fixed, but then it crashes anyway with:

Model building is not supported when publishing with NativeAOT. Use a compiled model.

```

	Basic Fault Address Reporting

Memory around native instruction pointer (0x7ff800ea6a57):0x7ff800ea6a47  c4 28 01 00 00 5b 41 5c 41 5d 41 5e 41 5f 5d c3  .(...[A\A]A^A_].
0x7ff800ea6a57  0f 0b e8 be cb 0f 00 55 48 89 e5 5d ff 25 4f 3a  .......UH..].%O:
0x7ff800ea6a67  96 67 55 48 89 e5 41 56 53 48 85 ff 74 3b e8 94  .gUH..AVSH..t;..
0x7ff800ea6a77  d3 0f 00 48 89 c7 e8 98 cd 0f 00 48 85 c0 74 29  ...H.......H..t)


	Managed Stacktrace:

	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:xamarin_UIApplicationMain <0x0019e>
	  at UIKit.UIApplication:UIApplicationMain <0x00122>
	  at UIKit.UIApplication:Main <0x004ca>
	  at Main:Main <0x000a2>
	  at <Module>:runtime_invoke_int_object <0x00110>
```

**Reply by maxkatz6** [reply]
> Model building is not supported when publishing with NativeAOT. Use a compiled model.

Do you have any specific/unusual configuration? Like enabling native aot on mobile?

**Reply by Aytharn** [reply]
Nope, it's just the Avalonia 10.0.5 template: I didn't change anything from it, except switching to FuncUI
<img width="914" alt="Screenshot 2023-12-14 at 21 11 13" src="https://github.com/fsprojects/Avalonia.FuncUI/assets/8307825/19dc9211-15e6-44f7-9637-b0d27505bd0d">

**Reply by Aytharn** [reply]
Could it be that it's Just EFCore8 not compatible with AOT under iOS ?
I never made it so far before, so I don't really know what's happening, I think the crash happens every time a view attempts to call EF

**Reply by maxkatz6** [reply]
> EFCore8 not compatible with AOT under iOS

I don't think EFCore is compatible with AOT anywhere.

> Nope, it's just the Avalonia 10.0.5 template: I didn't change anything from it, except switching to FuncUI


We don't have EFCore in default templates.

**Reply by maxkatz6** [reply]
cc @JaggerJo if you know where EFCore might come from.
